src/app/geonode/qgsgeonodesourceselect.cpp

```diff
@@ -40,10 +49,219 @@ QgsGeonodeSourceSelect::QgsGeonodeSourceSelect( QWidget *parent, Qt::WindowFlags
+}
+
+void QgsGeoNodeSourceSelect::setConnectionListPosition()
+{
+  QString toSelect = QgsGeoNodeConnection::selectedConnection();
+
+  cmbConnections->setCurrentIndex( cmbConnections->findText( toSelect ) );
+
+  if ( cmbConnections->currentIndex() < 0 )
+  {
+    if ( toSelect.isNull() )
+      cmbConnections->setCurrentIndex( 0 );
+    else
+      cmbConnections->setCurrentIndex( cmbConnections->count() - 1 );
+  }
+
+  if ( cmbConnections->count() == 0 )
+  {
+    // No connections - disable various buttons
+    btnConnect->setEnabled( false );
+    btnEdit->setEnabled( false );
+    btnDelete->setEnabled( false );
+    btnSave->setEnabled( false );
+  }
+  else
+  {
+    // Connections - enable various buttons
+    btnConnect->setEnabled( true );
+    btnEdit->setEnabled( true );
+    btnDelete->setEnabled( true );
+    btnSave->setEnabled( true );
+  }
+}
+
+void QgsGeoNodeSourceSelect::connectToGeonodeConnection()
+{
+  QgsGeoNodeConnection connection( cmbConnections->currentText() );
+  QVariantList layers = connection.getLayers();
+  QVariantList maps = connection.getMaps();
+
+  if ( mModel )
+  {
+    mModel->removeRows( 0, mModel->rowCount() );
+  }
+
+  if ( !layers.isEmpty() )
+  {
+    Q_FOREACH ( const QVariant &layer, layers )
+    {
+      QStandardItem *titleItem = new QStandardItem( layer.toMap()["title"].toString() );
+      QStandardItem *serviceTypeItem = new QStandardItem( "Feature Service" );
+      typedef QList< QStandardItem * > StandardItemList;
+      mModel->appendRow( StandardItemList() << titleItem << serviceTypeItem );
+    }
+  }
+
+  else
+  {
+    QMessageBox *box = new QMessageBox( QMessageBox::Critical, tr( "Error" ), tr( "Cannot get any feature services" ), QMessageBox::Ok, this );
+    box->setAttribute( Qt::WA_DeleteOnClose );
+    box->setModal( true );
+    box->setObjectName( QStringLiteral( "GeonodeCapabilitiesErrorBox" ) );
+    box->open();
+  }
+
+  if ( !maps.isEmpty() )
+  {
+    Q_FOREACH ( const QVariant &map, maps )
+    {
+      QStandardItem *titleItem = new QStandardItem( map.toMap()["title"].toString() );
+      QStandardItem *serviceTypeItem = new QStandardItem( "Map Service" );
+      typedef QList< QStandardItem * > StandardItemList;
+      mModel->appendRow( StandardItemList() << titleItem << serviceTypeItem );
+    }
+  }
+
+  else
+  {
+    QMessageBox *box = new QMessageBox( QMessageBox::Critical, tr( "Error" ), tr( "Cannot get any map services" ), QMessageBox::Ok, this );
+    box->setAttribute( Qt::WA_DeleteOnClose );
+    box->setModal( true );
+    box->setObjectName( QStringLiteral( "GeonodeCapabilitiesErrorBox" ) );
+    box->open();
+  }
+
+  treeView->resizeColumnToContents( MODEL_IDX_TITLE );
+  treeView->resizeColumnToContents( MODEL_IDX_TYPE );
+  for ( int i = MODEL_IDX_TITLE; i < MODEL_IDX_TYPE; i++ )
+  {
+    if ( treeView->columnWidth( i ) > 300 )
+    {
+      treeView->setColumnWidth( i, 300 );
+    }
+  }
+}
+
+void QgsGeoNodeSourceSelect::on_btnSave_clicked()
```

Comment:
Yes, this is legacy. QGIS Developers are removing them step by step.
Having an explicit connection is better.